_posts/2022-10-05-pdf-generation-at-scale.md

```diff
@@ -0,0 +1,212 @@
+---
+title: "PDF generation at scale with OpenFaaS and Puppeteer"
+description: "We show how to use OpenFaaS with back pressure and autoscaling to generate PDFs with puppeteer"
+date: 2022-10-05
+categories:
+- architecture
+- casestudy
+- chrome
+- automation
+author_staff_member: han
+---
+
+We show how to use OpenFaaS with back pressure and autoscaling to generate PDFs with puppeteer.
+
```

Comment:
> Intro from Alex:

> We had a call with a Kubernetes user at a well known start-up who used to run PDF generation on AWS Lambda. They told us that it worked flawlessly and despite its cost, scaled exactly how they wanted. After moving to Kubernetes, they started to run into various problems scaling headless Chrome and it's still a pain point for them today.

> This article shows how we approached the problem using built-in features of OpenFaaS - connection-based (aka capacity), hard limits (aka max_inflight) and a flexible queuing system that can retry failed invocations.

> If you follow the instructions, you'll be able to scale out headless chrome for e2e testing, PDF generation or any other kind of automation that you need, without worrying about reliability.

Reply:
Do you want to use this as the intro are as a separate "intro from Alex"?